### PR TITLE
Check if email is settled for GDPR export and delete

### PR DESCRIPTION
**Pull request by clementdaubeuf**
Updated version of this [PR](https://github.com/PrestaShop/ps_emailsubscription/pull/30)

### CODE REVIEW

File: ps_emailsubscription.php [hunk 1]
```diff
@@ -1353,25 +1353,27 @@ public function hookDisplayAdminCustomersForm()
 
     public function hookActionDeleteGDPRCustomer($customer)
     {
-        if (!empty($customer['email']) && Validate::isEmail($customer['email'])) {
-            $sql = 'DELETE FROM ' . _DB_PREFIX_ . "emailsubscription WHERE email = '" . pSQL($customer['email']) . "'";
-            if (Db::getInstance()->execute($sql)) {
-                return json_encode(true);
+        if (isset($customer['email'])) {
```
Comment (Progi1984) [inline ps_emailsubscription.php:1356]: `isset` is not useful because the `!empty` on the line beside do the job.

File: ps_emailsubscription.php [hunk 2]
```diff
@@ -1353,25 +1353,27 @@ public function hookDisplayAdminCustomersForm()
...
     }
 
     public function hookActionExportGDPRData($customer)
     {
-        if (!Tools::isEmpty($customer['email']) && Validate::isEmail($customer['email'])) {
-            $sql = 'SELECT * FROM ' . _DB_PREFIX_ . "emailsubscription WHERE email = '" . pSQL($customer['email']) . "'";
-            if ($res = Db::getInstance()->ExecuteS($sql)) {
-                return json_encode($res);
+        if (isset($customer['email'])) {
+            if (!Tools::isEmpty($customer['email']) && Validate::isEmail($customer['email'])) {
```
Comment (Progi1984) [inline ps_emailsubscription.php:1370]:
```suggestion
            if (!empty($customer['email']) && Validate::isEmail($customer['email'])) {
```

Comment (clementdaubeuf) [reply ps_emailsubscription.php]: I implemented your suggestion, thank you for your help :)

File: ps_emailsubscription.php (hunk 1 above)
Comment (Progi1984) [reply ps_emailsubscription.php:1356]: Please remove isset, like empty is used

Comment (Progi1984) [reply ps_emailsubscription.php:1356]: Ping @clementdaubeuf

**Comment by seniordelosgatos** [comment]
Hello,
One addition to this function: it only check if email is in the guests newsletter table. For registered users no subscription is checked:

$sql = 'SELECT * FROM ' . _DB_PREFIX_ . "emailsubscription WHERE email = '" . pSQL($customer['email']) . "'";

When a customer suscribes newsletter as a guest, the email address inserts in table "emailsubscription" but when is a registered customer this table is not used, because the info from customers table is used.

This function should use same logic as this (sorry, I'm not a developer and can't put more accurate example):

    public function isNewsletterRegistered($customer_email)
    {
        $sql = 'SELECT `email`
                FROM ' . _DB_PREFIX_ . 'emailsubscription
                WHERE `email` = \'' . pSQL($customer_email) . '\'
                AND id_shop = ' . $this->context->shop->id;

        if (Db::getInstance()->getRow($sql)) {
            return self::GUEST_REGISTERED;
        }

        $sql = 'SELECT `newsletter`
                FROM ' . _DB_PREFIX_ . 'customer
                WHERE `email` = \'' . pSQL($customer_email) . '\'
                AND id_shop = ' . $this->context->shop->id;

        if (!$registered = Db::getInstance()->getRow($sql)) {
            return self::GUEST_NOT_REGISTERED;
        }

        if ($registered['newsletter'] == '1') {
            return self::CUSTOMER_REGISTERED;
        }

        return self::CUSTOMER_NOT_REGISTERED;

**Comment by Progi1984** [comment]
Ping @clementdaubeuf

**Comment by matks** [comment]
Hello, it seems this PR was not updated since 1 year.

I close this PR, following our [review guidelines](https://devdocs.prestashop.com/1.7/project/maintainers-guide/reviewing-pull-requests/) "Pull Requests may be closed after 30 days of inactivity". If however you are still interested in it, please ping us :) we can reopen.

Thank you for this contribution

**Pull request closed without merging**